KeyCloak service migration doc (#177)
BCDevOps/developer-experience: docs/kcMigration/KeycloakMigration.md

---
description: KeyCloak (RH-SSO) Migration from Pathfinder to BCGov service
resourceType: Documentation
title: BC Government KeyCloak (RH-SSO) Service Migration Plan

> ---

Reviewer: suggest adding tags for searchability like
```
tags:
- pathfinder migration
- keycloak host name etc

## KeyCloak Service Migration

> Hi team, we are in the process of transforming KeyCloak from a pathfinder service to an organizational BCGov service. For more information on this Single Sign-On service, refer to [here](https://developer.gov.bc.ca/BC-Government-SSO-Service-Definition).

Reviewer: 1. I'd avoid sending this as a personal letter using words like 'we'. Just tell them how it is. 
Something like 'Keycloak is being converted to an BC Gov organizational service. This means it is being migrated from the `pathfinder` service.

refer to [BC Government SSO Service Definition]..

> Part of the migration requires you (application teams using this existing SSO) to update the KeyCloak configurations. As for now, there is no immediate actions to migrate. We will inform the community in [RocketChat #sso channel](https://chat.pathfinder.gov.bc.ca/channel/sso) as well as product owners when ready so that you could plan ahead for some sprint time to work on the relevant tasks.

Reviewer 2: ```suggestion
```
Can move further down and lead with what you need actioned right away.

> At the moment, we are collecting application team information for each KC realm. Please visit the rocketchat discussion [here](to-be-generated), find your realm and provide the details listed:

Reviewer 2: ```suggestion
The first step of the migration involving you (Application teams using the existing SSO) is the collection of project team contact information for each KC realm.  Please visit the Rocket.Chat discussion [here](to-be-generated), find your realm and provide the details listed:
```

- Project Name: 
- KC realm ID: 
- OpenShift project set/s: 

> If your team is not on the OpenShift platform, please provide contact information for:

Reviewer 2: ```suggestion
If your team is not on the OpenShift platform, please also provide contact information for:
```

- Product owner: 
- Technical lead: 

> 

Reviewer 2: ```suggestion
At this point, there are no immediate actions required to migrate.  Once the new service name(s) are in place, you will be required to update your realm's KeyCloak configurations as well as any application links to the keycloak DNS.  We will inform the community in [RocketChat #sso channel](https://chat.pathfinder.gov.bc.ca/channel/sso) as well as product owners as the start date approaches so that you can plan ahead for some sprint time to work on the relevant tasks.

```
To give the teams the heads up of more work to come.

> ### Migration plan and phases:

Reviewer: I think here we should identify who are involved in each phases as personas. Perhaps in bold?

Phase 1. __Platform Services and Platform Operations__ will test out

Phase 4. Work with __Pilot Teams__ to..

---
description: KeyCloak (RH-SSO) Migration from Pathfinder to BCGov service
resourceType: Documentation
title: BC Government KeyCloak (RH-SSO) Service Migration Plan
tags:
- sso service migration
- keycloak host name change
---

## KeyCloak Service Migration

> Hi team, the pathfinder KeyCloak SSO service is undergoing a transformation to an organizational BCGov service. For more information on this Single Sign-On service, refer to [here](https://developer.gov.bc.ca/BC-Government-SSO-Service-Definition).

Reviewer 2: You are extremely vague about what that transformation is, and how it will affect the users of the service.  Would be awesome if you could be more specific.. maybe:
```suggestion
Hi team, the KeyCloak SSO service is finally removing its *Pathfinder* sticker and getting a new name.  This change does not affect the service offering but does change how you will access it.  The name is changing from sso.pathfinder.gov.bc.ca to oidc.gov.bc.ca.  For more information on this Single Sign-On service, refer to [here](https://developer.gov.bc.ca/BC-Government-SSO-Service-Definition).
```
